src/platform/unix/mod.rs

```diff
@@ -79,6 +81,21 @@ pub fn channel() -> Result<(OsIpcSender, OsIpcReceiver),UnixError> {
     }
 }
 
+#[derive(Clone, Copy)]
+struct FDHandle {
+    pub id: u64,
+    pub descriptor: c_int
+}
```

Comment:
The `OsIpcReceiverSet::pollfds` and `OsIpcReceiverSet::fdids` `HashMaps` have a 1:1 relationship. The `FDHandle` structure contains the information that would previously be contained in the two.